src/provider/systemdmanager/systemdunit.rs

```diff
@@ -64,39 +127,79 @@ impl SystemDUnit {
 
         unit.name = format!("{}{}", name_prefix, trimmed_name);
 
-        unit.add_property(SECTION_UNIT, "Description", &unit.name.clone());
+        unit.add_property(Section::Unit, "Description", &unit.name.clone());
 
         unit.add_property(
-            SECTION_SERVICE,
+            Section::Service,
             "ExecStart",
-            &SystemDUnit::get_command(container, pod_state)?,
+            &SystemDUnit::get_command(container, template_data, package_root)?,
         );
 
-        let env_vars = SystemDUnit::get_environment(container, pod_state)?;
+        unit.add_property(
+            Section::Service,
+            "Environment",
+            &SystemDUnit::get_environment(container, service_name, template_data)?
+                .iter()
+                .map(|(k, v)| format!("\"{}={}\"", k, v))
+                .collect::<Vec<_>>()
+                .join(" "),
+        );
+
+        // These are currently hard-coded, as this is not something we expect to change soon
+        unit.add_property(Section::Service, "StandardOutput", "journal");
+        unit.add_property(Section::Service, "StandardError", "journal");
 
-        for (name, value) in env_vars {
-            unit.add_env_var(&name, &value);
+        if let Some(user_name) =
+            SystemDUnit::get_user_name_from_security_context(container, &unit.name)?
+        {
+            if !user_mode {
```

Comment:
In principle I agree, but knowing users ... no one reads the docs..
I pretty much don't care whether it is warn or info, but I to think we should log this, because we deviate from the intent that was expressed in the cluster definition by the use. We do it because of something else the user did, that is true, but that could have been a totally different person and we want to be as predictable as possible - strictly speaking, if we follow our "fail early" principle I'd say we should fail that pod, because we cannot run it as specified..

paging @lfrancke  for a third (and probaby entirely different) opinion